docs/source/type_compatibility.rst

```diff
@@ -0,0 +1,17 @@
+******************
+Type Compatibility
+******************
+
+
+The Class Hierarchy
+===================
+
+Mutable Containers
```

Comment:
It might make sense to move the section to where we introduce generics, since it's hard to have any discussion of variance without examples.